WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+function CompetitionTable({
+  competitions,
+  title,
+  sortByAnnouncement = false,
+}) {
+  return (
+    <ul className="list-group">
+      <li className="list-group-item">
+        <strong>
+          {`${title} (${competitions.length})`}
+        </strong>
+      </li>
+      {competitions.map((comp, index) => (
+        <>
+          {(index > 0 && comp.year !== competitions[index - 1].year && !sortByAnnouncement) && <li className="list-group-item break">{comp.year}</li>}
```

Comment:
Extracting this condition to a helper function `isFirstCompOfAYear(competitions, index)` would make it easier to understand.